src/ppl/src/HungarianMatching.cpp

```diff
@@ -119,10 +124,33 @@ void HungarianMatching::getFinalAssignment(std::vector<IOPin>& assigment) const
                         "Not enough space.",
                         io_pin.getName().c_str());
         }
+
+        // Make this check here to avoid messing up the correlation between the
+        // pin sorting and the hungarian matrix values
+        if ((assign_mirrored
+             && mirrored_pins.find(io_pin.getBTerm()) == mirrored_pins.end())
+            || io_pin.isPlaced()) {
+          continue;
+        }
         io_pin.setPos(slots_[slot_index].pos);
         io_pin.setLayer(slots_[slot_index].layer);
+        io_pin.setPlaced();
         assigment.push_back(io_pin);
         slots_[slot_index].used = true;
+
+        if (assign_mirrored) {
```

Comment:
What happens if the mirrored slot is already in use?

Reply:
I thought this scenario could be avoided by first assigning the mirrored pins and then the remaining pins. This way, all the slots are free when assigning the mirrored slots.

But I just realized that when we have pin groups, we can have used slots when assigning the mirrored pins. We are currently given priority for groups over the remaining pins. I think prioritizing the mirrored pins over the groups should work, but I wonder how often we'll have pin groups and mirrored pins together.

Reply:
@oharboe do you plan to have mirrored busses?

Reply:
I dont know what a mirrored bus is.

Reply:
Will you have a group of pins that you also want to mirror?

Reply:
I am trying to do "routing by abutment", i.e. that all the routing happens inside the macros and that I can just place the macros next to eachother with no spacing and have them connect correctly. So all the pins on one edge.must match exactly all the pins on the neighbouring edge.

Reply:
I have no opinion on the order of the pins, but it is probably worth ordering them. Easier to view it in the GUI. Faster to debug.

I am doing design exploration, not tapeout, so I am more concerned about implementation effort and the pedagogic element than sqeueezing out the last drops of PPA.